version.go

```diff
@@ -14,7 +14,11 @@ func version() string {
 	vOnce.Do(func() {
 		bi, ok := debug.ReadBuildInfo()
 		if ok {
-			v = bi.Main.Version
+			for _, dep := range bi.Deps {
+				if dep.Path == "github.com/IBM/sarama" {
```

Comment:
🤔 huh… it would probably be way better if we had some sort of “give me the build info for this module itself” rather than relying on the name here… otherwise this becomes a brittle magic string.

Actually… maybe this might work?: https://go.dev/play/p/LC1Q8tpP83T
```go
type getPackageName struct{}
thisPackagePath := reflect.TypeFor[getPackageName]().PkgPath()
```

Reply:
That's a good idea. I've updated the PR.

Reply:
I've verified this works when compiled into a binary as well. Using `replace` to point to a local copy of IBM/sarama still finds `github.com/IBM/sarama` and `v1.46.3` when running as a library in a binary.

Reply:
Hmm. The reflect approach adds an extra import, a dummy struct type and a reflect call at runtime and the only benefit would come if the module was ever renamed in the future – but that rename would already comprise rewriting all the import path strings anyway, so it’s not immediately obvious to me that it’s preferable to the simple hardcoded string comparison that you originally had?